packages/nimble-components/src/rich-text-viewer/tests/rich-text-viewer.spec.ts

```diff
@@ -10,4 +29,209 @@ describe('RichTextViewer', () => {
+            expect(childElement!.getAttribute('href')).toBe(
+                'https://nimble.ni.dev/'
+            );
+
+            await disconnect();
+        });
+    });
+
+    describe('text formatting styles apart from the supported styles should be considered as a paragraph tag with a plain text', () => {
+        async function expectParagraphTag(
+            markdownValue: string
+        ): Promise<void> {
+            element.markdownValue = markdownValue;
+
+            await connect();
+
+            const childElement = pageObject.getFirstChildElement();
+            expect(pageObject.getTagName(childElement)).toBe('P');
+            expect(pageObject.getTextContent(childElement)).toBe(markdownValue);
+
+            await disconnect();
+        }
+
+        it('should be rendered as a paragraph tag for "blockquote" markdown string', async () => {
+            const markdownValue = '> blockquote';
+            await expectParagraphTag(markdownValue);
+        });
+
+        it('should be rendered as a paragraph tag for "code" markdown string', async () => {
+            const markdownValue = '`code`';
+            await expectParagraphTag(markdownValue);
+        });
+
+        it('should be rendered as a paragraph tag for "fence" markdown string', async () => {
+            const markdownValue = '```fence```';
+            await expectParagraphTag(markdownValue);
+        });
+
+        it('should be rendered as a paragraph tag for "heading" markdown string', async () => {
+            const markdownValue = '# Heading';
+            await expectParagraphTag(markdownValue);
+        });
+
+        it('should be rendered as a paragraph tag for "hyperlink" markdown string', async () => {
+            const markdownValue = '[line](url)';
+            await expectParagraphTag(markdownValue);
+        });
+
+        it('should be rendered as a paragraph tag for "reference" markdown string', async () => {
+            const markdownValue = '[ref][link] [link]:url';
+            await expectParagraphTag(markdownValue);
+        });
+
+        it('should be rendered as a paragraph tag for "image" markdown string', async () => {
+            const markdownValue = '![Text](Image)';
+            await expectParagraphTag(markdownValue);
+        });
+
+        it('should be rendered as a paragraph tag for "entities" markdown (string starting with "&" symbol)', async () => {
+            const markdownValue = '&nbsp;';
+            await expectParagraphTag(markdownValue);
+        });
+
+        it('should be rendered as a paragraph tag for "horizontal rule" markdown string', async () => {
+            const markdownValue = '---';
+            await expectParagraphTag(markdownValue);
+        });
+
+        it('should be rendered as a paragraph tag for HTML tags', async () => {
+            const markdownValue = '<div><p>text</p></div>';
+            await expectParagraphTag(markdownValue);
+        });
```

Comment:
For these tests all the text execution and expects are the same, they just need different input parameters. To handle that you should use the `getSpecTypeByNamedList` helper instead to make parametrized tests.

See the following test as an example:

https://github.com/ni/nimble/blob/12463598a33340eb311356d29b49d1195d228d86/packages/nimble-components/src/anchor-tab/tests/anchor-tab.spec.ts#L30-L63

Reply:
Actually that might not be the best example, just know that the array can have multiple parameters as needed, like`[{name: "blah", config1: "some val", config2: 5}]`. Look for other examples of `getSpecTypeByNamedList` as needed.

Reply:
I have updated these test cases using `getSpecTypeByNamedList`. For these tests, I couldn't think of any other parameters that need to be added to the array so I added only the not supported markdown strings for now.